server/server.go

```diff
@@ -36,19 +52,101 @@ func upload(w http.ResponseWriter, r *http.Request) {
 	w.Write(responseS)
 }
 
-// Reads the compressed file and invokes function to decompress it
+// Builds and runs the assignment uploaded.
+func buildRun(w http.ResponseWriter, r *http.Request) {
+
+	fmt.Println("Build and Run Endpoint Hit")
+	w.Header().Set("Access-Control-Allow-Origin", "*")
+
+	var outputString string
+	var currDir string
+	var err error
+
+	outputString, currDir = navigateToWorkDir()
+	if outputString == "" {
+		outputString, err = runCommand(AssignmentData.CommandToCompile)
+
+		if err == nil {
+			finalCmd := AssignmentData.CommandToExecute
+			for _, value := range AssignmentData.CmdlineArgs {
+				finalCmd = fmt.Sprintf("%s %s", finalCmd, value)
+			}
+			fmt.Println(finalCmd)
+			outputString, err = runCommand(finalCmd)
+		}
+
+		err = os.Chdir(currDir)
+		if err != nil {
+			log.Fatalf("error while navigating to the current directory: %v", err)
```

Comment:
these log.Fatal's can leave the system in an unsafe state.
Always perform graceful shutdown.
Raise an issue for this.

Reply:
Okay